topology/probes/netlink_unspecified.go

```diff
@@ -0,0 +1,62 @@
+// +build !linux
+
+package probes
+
+import (
+	"errors"
+	"net"
+	"sync"
+
+	"github.com/skydive-project/skydive/topology/graph"
+)
+
+var (
+	ErrNotImplemented = errors.New("not implemented")
+)
+
+// NetNsNetLinkProbe describes a topology probe based on netlink in a network namespace
+type NetNsNetLinkProbe struct {
```

Comment:
It would be better to have common files like netlink.go and specific files for unix and unspecified like netlink_unix.go and netlink_unspecified.go in order to avoid having structs duplicated. Same thing for the other files.

Reply:
This makes sense for the constants in options.go.  For the others, it makes sense to rename the files with _linux.go.  (The structs will mostly still be duplicated, as these structs all contain private members that depend on things that are not present on OSX, or implemented in a different way).

Reply:
For the selenium test nothing related to your patch, I just restarted it

Reply:
We think that if the goal is to be able to compile only the client part of Skydive it would be better to isolate the client. Otherwise you will get a binary with an agent and an analyzer. Moreover we would prefer to be able to cross compile the client so that we will be able to test it otherwise it is pretty sure that at some point we will break the MacOs support.

@lebauce has a draft which modifiers skydive that way. Can you please test it ?

Reply:
https://github.com/lebauce/skydive is old and didn't build for me.

I am modifying the PR so that the binary does not include agent and analyzer on OSX.  This change reduced the OSX _skydive_ binary from 78MB to 62MB.

This PR does three things.  1. change Makefile _sed_ flags which helps with OSX (and maybe BSD.)  Brings in a newer version of vishvananda/netns which is needed anyway.  3. Makes a bunch of methods return "unimplemented".  (Maybe someone with more networking skill will implement for these OSX or use them to get started for another OS.)

I have not cross-compiled a Go program with "// build" comments but suspect it will work as expected.  You should have no trouble building OSX binaries on Linux with these changes, but without them you won't be able to cross-compile an OSX client binary on Linux.

It is possible in the future someone might break skydive into a client, an agent/analyzer, and a common library.  That is beyond my skill level.